src/silx/gui/plot/LegendItem.py

```diff
@@ -0,0 +1,156 @@
+        ):
+            self._update()
+
+    def getItem(self) -> items.Item | None:
+        return self._item
+
+    def _update(self):
+        item = self.getItem()
+        if item is None:
+            _logger.debug("Item no longer exists, disabling legend widget.")
+            self.setEnabled(False)
+            return
+
+        self._icon.setEnabled(item.isVisible())
+        if isinstance(item, items.SymbolMixIn):
+            self._icon.setSymbol(item.getSymbol())
+        if isinstance(item, items.LineMixIn):
+            self._icon.setLineWidth(item.getLineWidth())
+            self._icon.setLineStyle(item.getLineStyle())
+        if isinstance(item, items.ColorMixIn):
+            color_rgba = item.getColor()
+            color = qt.QColor.fromRgbF(
+                color_rgba[0],
+                color_rgba[1],
+                color_rgba[2],
+                color_rgba[3],
+            )
+            self._icon.setLineColor(color)
+            self._icon.setSymbolColor(color)
+        if isinstance(item, items.ColormapMixIn):
+            self._icon.setColormap(item.getColormap())
+
+        self._label.setText(item.getName())
+
+        style = ""
+        palette = self.palette()
+        text_color = palette.color(qt.QPalette.ColorRole.Text).name()
+        disabled_text_color = (
+            palette.color(qt.QPalette.ColorRole.Text).darker(150).name()
+        )
+
+        if isinstance(item, HighlightedMixIn) and item.isHighlighted():
+            style += "border: 1px solid black; "
+
+        if not item.isVisible():
+            style += f"color: {disabled_text_color};"
+        else:
+            style += f"color: {text_color};"
+
+        self._label.setStyleSheet(style)
+
+    def _onLabelClicked(self, event):
+        if event.button() == qt.Qt.LeftButton:
+            if self.getItem():
+                self.getItem().setVisible(not self._item.isVisible())
+
+
+class LegendItemList(qt.QWidget):
+    def __init__(
+        self,
+        parent: qt.QWidget | None = None,
+        plotWidget: PlotWidget | None = None,
+    ):
+        super().__init__(parent)
+        self.setLayout(qt.QVBoxLayout(self))
+        self.setSizePolicy(
+            qt.QSizePolicy.Policy.Preferred, qt.QSizePolicy.Policy.Expanding
+        )
+        self.setMinimumWidth(150)
+        self.setMaximumWidth(300)
+
+        self._plot: PlotWidget | None = plotWidget
+        self._binding(plotWidget=plotWidget)
+        self._itemWidgets = {}
+
+    def _binding(self, plotWidget: PlotWidget):
+        """Binds this widget to the signals of a parent PlotWidget."""
+        self._plot = plotWidget
+        self._plot.sigItemAdded.connect(self._updateAllItemsList)
+        self._plot.sigItemRemoved.connect(self._onItemRemoved)
+        self._updateAllItemsList()
+        self.show()
+
+    def _clearLayout(self, layout):
+        """Helper to clear all widgets from a layout."""
+        if layout is not None:
```

Comment:
No need to pass the layout as an argument:
```suggestion
    def _clear(self):
        """Helper to clear all widgets from a layout."""
        layout = self.layout()
        if layout is not None:
```
+ this does not clear `self._itemWidgets`